src/bot_telegram.ts

```diff
@@ -154,13 +154,29 @@ export default class TelegramBot extends BotClient {
     // Image Links
     markdown = markdown.replace(/\[\!\[\]\((.*)\)\]\((.*)\)/g, '[Image]($1) ([Link]($2))');
     markdown = markdown.replace(/\[\!\[(.*)\]\((.*)\)\]\((.*)\)/g, '[$1]($2) ([Link]($3))');
-    markdown = markdown.replace(/\!\[\]\((.*)\)/g, '[Image] ($1)');
-    markdown = markdown.replace(/\!\[(.*)\]\((.*)\)/g, '[$1] ($2)');
+    markdown = markdown.replace(/\!\[\]\((.*)\)/g, '[Image]($1)');
+    markdown = markdown.replace(/\!\[(.*)\]\((.*)\)/g, '[$1]($2)');
     // Italic
     markdown = markdown.replace(/\*(?!\*)(.+)(?!\*)\*/g, '_$1_');
     // Bold
     markdown = markdown.replace(/__(.+)__/g, '*$1*');
     markdown = markdown.replace(/\*\*(.+)\*\*/g, '*$1*');
+    // Formatting with Links
+    markdown = markdown.replace(/\*(.*)[ \t]*\[(.*)\]\((.*)\)[ \t]*(.*)\*/g, '*$1* [$2]($3) *$4*');
+    markdown = markdown.replace(
+      /__(.*)[ \t]*\[(.*)\]\((.*)\)[ \t]*(.*)__/g,
+      '__$1__ [$2]($3) __$4__',
+    );
+
+    markdown = markdown.replace(/\[\*(.*)\*\]\((.*)\)/g, '[$1]($2)');
+    markdown = markdown.replace(/\[__(.*)__\]\((.*)\)/g, '[$1]($2)');
```

Comment:
I can see some regular expressions being reused. Do you think it would make sense to standardize them, and perhaps even split them to meaningful components which can be combined to make a specific regex? 
For example: `(.*)` is pretty standard and could be named something and reused. 

Not very sure about the splitting part, but naming regexes sounds like a good idea. For example we can have:
``` typescript
linkRegex = /\[\*(.*)\*\]\((.*)\)/g;
markdown = markdown.replace(linkRegex, '[$1]($2)');
```
or something along these lines

Reply:
That's a good point and would probably make bugfixing easier as well.
However, I don't think you can put a regex in another regex, so we would have to use strings for everything and then convert those to regexes in the replace method. But that's probably still better than the current approach.

Reply:
I will fix this in a seperate issue (#60).